host_core/lib/host_core/control_interface/server.ex

```diff
@@ -586,4 +587,8 @@ defmodule HostCore.ControlInterface.Server do
       error: error
     })
   end
+
+  defp has_keys?(keys, request) do
+    Enum.all?(keys, &Map.has_key?(request, &1))
+  end
```

Comment:
Just for safety, since the `request` is user input here, I'd recommend adding a `when is_map(request)` to this function and setting up another function that handles the fallback case
```
defp has_keys?(keys, request), do: false
```